src/main/java/seedu/clinic/logic/parser/ViewCommandParser.java

```diff
@@ -1,41 +1,88 @@
 
+    private final Logger logger = LogsCenter.getLogger(getClass());
+
     @Override
     public ViewCommand parse(String args) throws ParseException {
-        String trimmedArgs = args.trim();
-        if (trimmedArgs.isEmpty()) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, ViewCommand.MESSAGE_USAGE));
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_TYPE, PREFIX_INDEX);
+
+        logger.log(Level.INFO, "Tokenised user inputs.");
+
+        if (!argMultimap.getValue(PREFIX_TYPE).isPresent()
+                && !argMultimap.getValue(PREFIX_INDEX).isPresent()) {
+            throw new ParseException(String.format(MESSAGE_NO_PREFIX, ViewCommand.MESSAGE_USAGE));
         }
 
-        String[] userInputs = trimmedArgs.split("\\s+");
-        if (userInputs.length < NUMBER_OF_ARGUMENTS) {
+        if (!argMultimap.getValue(PREFIX_TYPE).isPresent()) {
+            throw new ParseException(String.format(MESSAGE_MISSING_TYPE, ViewCommand.MESSAGE_USAGE));
+        }
+
+        if (!argMultimap.getValue(PREFIX_INDEX).isPresent()) {
             throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, ViewCommand.MESSAGE_TOO_FEW_ARGUMENTS));
+                    String.format(MESSAGE_MISSING_INDEX, ViewCommand.MESSAGE_USAGE));
+        }
+
+        Type type;
+        try {
+            type = ParserUtil.parseType(argMultimap.getValue(PREFIX_TYPE).get());
+        } catch (ParseException pe) {
+            throw checkInvalidArguments(PREFIX_TYPE, argMultimap);
         }
 
-        String type = userInputs[TYPE_IN_VIEW_COMMAND].toLowerCase();
-        List<String> name = Arrays.asList(userInputs).subList(NAME_IN_VIEW_COMMAND, userInputs.length);
+        logger.log(Level.INFO, "Successfully parsed command type of user input.");
 
-        if (!Arrays.asList(ALLOWED_TYPES).contains(type.toLowerCase())) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, ViewCommand.MESSAGE_INVALID_TYPE));
+        if (type.equals(Type.SUPPLIER_PRODUCT) || type.equals(Type.WAREHOUSE_PRODUCT)) {
+            throw new ParseException(String.format(MESSAGE_INVALID_TYPE_VIEW, ViewCommand.MESSAGE_USAGE));
         }
 
-        return new ViewCommand(type, name);
+        Index index;
+
+        try {
+            index = ParserUtil.parseIndex(argMultimap.getValue(PREFIX_INDEX).get());
+        } catch (ParseException pe) {
+            throw checkInvalidArguments(PREFIX_INDEX, argMultimap);
+        }
+
+        logger.log(Level.INFO, "Successfully parsed index of user input, creating new ViewCommand.");
+
+        return new ViewCommand(type, index);
+    }
+
+    private ParseException checkInvalidArguments(Prefix prefix, ArgumentMultimap argMultimap) {
+        if (argMultimap.getValue(prefix).get().contains("/")) {
```

Comment:
Have to discuss this later, how to handle it efficiently since not every command takes in email/address/name and like in FindCommand we have to add this to check for "/".